pkg/router/controller/router_controller.go

```diff
@@ -321,11 +327,13 @@ func convertEndpointSliceToEndpointSubset(items []discoveryv1beta1.EndpointSlice
 		}
 
 		endpointsubset.SortAddresses(addresses, addressOrderByFuncs)
+		endpointsubset.SortAddresses(notReadyAddresses, addressOrderByFuncs)
 		endpointsubset.SortPorts(ports, portOrderByFuncs)
 
 		subsets = append(subsets, kapi.EndpointSubset{
-			Addresses: addresses,
-			Ports:     ports,
+			Addresses:         addresses,
+			NotReadyAddresses: notReadyAddresses,
```

Comment:
It doesn't look like the router actually uses `NotReadyAddresses`...  But we can do this to have a faithful and technically correct translation of EndpointSlices to Endpoints.

Reply:
Added this comment to the PR description. Thanks!